CODINGSTYLE.md

```diff
@@ -0,0 +1,335 @@
+# Coding Style Guide
+
+The Scala.js project has pretty drastic rules in terms of coding style.
```

Comment:
That's an honorable effort to be kind, but saying it the way you do would be a plain lie ^^

> to make it easier for everyone to contribute

Given how contributors first react to our dedication to coding style, I hardly believe strict coding style to make it easier for everyone to contribute. Do you honestly believe that? Besides, it really is not the point; the point is to make it easier to *maintain* the codebase, which is something most contributors never do.

> contributions are highly encouraged to follow them

This also does not do justice to the fact that we, in fact, don't accept a PR until it meets the standard.

That said, I agree my sentence is probably *too* harsh. What about this:
"The Scala.js project has a strict policy regarding coding style. This document tries to document the style in use as much as possible."
?